### PR TITLE
gui-apps/waybar: New package (0.7.2, 9999)

### PR DESCRIPTION
**Pull request by lovesegfault**
Closes: https://bugs.gentoo.org/684708
Package-Manager: Portage-2.3.70, Repoman-2.3.16
Signed-off-by: Bernardo Meurer <bernardo@standard.ai>

### CODE REVIEW

File: gui-apps/waybar/waybar-0.7.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+	libcxx? ( sys-libs/libcxx )
+	mpd? ( media-libs/libmpdclient )
+	network? ( dev-libs/libnl:3 )
+	pulseaudio? ( media-sound/pulseaudio )
+	tray? ( dev-libs/libdbusmenu[gtk3] )
+	udev? ( virtual/libudev )
+"
+RDEPEND="
+	${DEPEND}
```
Comment (lanodan) [inline gui-apps/waybar/waybar-0.7.1.ebuild]: You should put `:=` slot rules on dependencies it's linking with and that have soname versionning. For example jsoncpp with `jsoncpp:0/19` providing `libjsoncpp.so.19`.

Comment (lovesegfault) [reply gui-apps/waybar/waybar-0.7.1.ebuild]: How do I figure out the correct soname version? I tried listing all the linked libraries on my exec, but I'm not sure I should use it as a reference: https://gist.github.com/lovesegfault/88e23dccb2aa0bdea0d3cccf664d0318

**Comment by lovesegfault** [comment]
@gyakovlev Apparently I missed this during our chat on IRC, but the live version of this does not build, due to an inconsistent source directory (the capitalization problem again.) What's the analogue of `S` for the live case?

**Comment by gyakovlev** [comment]
You can define `S` right after keywords in conditional block. This will keep versioned build happy and skip applying for 9999.

**Comment by lovesegfault** [comment]
Got it and fixed, thank @gyakovlev!

File: gui-apps/waybar/waybar-0.7.1.ebuild (hunk 1 above)
Comment (lanodan) [reply gui-apps/waybar/waybar-0.7.1.ebuild]: IMHO just putting a `:=` on the existing deps that you have should be quite okay, specially if they end in `.0` and/or have no (sub)slot in gentoo yet.

Comment (lovesegfault) [reply gui-apps/waybar/waybar-0.7.1.ebuild]: Done! :)

Comment (flappyports) [reply gui-apps/waybar/waybar-0.7.1.ebuild]: subslots trigger rebuilds should the soname change from the package providing it.  That is the most important factor.

File: gui-apps/waybar/metadata.xml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        <name>Gentoo Proxy Maintainers Project</name>
+    </maintainer>
+    <longdescription lang="en">
+        Highly customizable Wayland bar for Sway and Wlroots based compositors.
+    </longdescription>
+    <use>
+        <flag name="libcxx">Build with Clang's libc++ instead of libstdc++ on Linux.</flag>
+        <flag name="mpd">Enable support for the Music Player Daemon</flag>
+        <flag name="network">Enable libnl support for network related features</flag>
+        <flag name="pulseaudio">Enable support for pulseaudio</flag>
```
Comment (flappyports) [inline gui-apps/waybar/metadata.xml]: pulseaudio is a global USE.

File: gui-apps/waybar/metadata.xml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        <flag name="tray">Enable support for tray</flag>
+        <flag name="udev">Enable libudev support for udev related features</flag>
```
Comment (flappyports) [inline gui-apps/waybar/metadata.xml]: udev is a global USE.

File: gui-apps/waybar/waybar-0.7.1.ebuild [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/Alexays/${PN^}.git"
+else
+	SRC_URI="https://github.com/Alexays/${PN^}/archive/${PV}.tar.gz -> ${P}.tar.gz"
+	KEYWORDS="~amd64"
+fi
+
+LICENSE="MIT"
+SLOT="0"
+IUSE="libcxx mpd network pulseaudio tray udev"
```
Comment (flappyports) [inline gui-apps/waybar/waybar-0.7.1.ebuild]: I believe +udev would be a sane default here.

Comment (lovesegfault) [reply gui-apps/waybar/waybar-0.7.1.ebuild]: Changed! Thanks!

**Review by kidomangp — APPROVED** [review]

**Review by kidomangp — APPROVED** [review]
Link

**Pull request closed without merging**